src/components/checkboxTree/components/RenderComposition.tsx

```diff
@@ -46,6 +47,14 @@ export function RenderComposition({
   };
 
   React.useEffect(() => {
+    if (
+      parentUid &&
+      selectedList.includes(parentUid) &&
+      !selectedList.includes(uid)
+    ) {
+      setSelectedList([...selectedList, uid]);
```

Comment:
I guess we don't need to evaluate another `if`, right?
```suggestion
      setSelectedList([...selectedList, uid]);
      return
```